File: sample-app/src/main/java/dev/oneuiproject/oneuiexample/ui/main/fragment/WidgetsFragment.java
package dev.oneuiproject.oneuiexample.ui.main.fragment;

import static android.view.View.VISIBLE;

import android.app.SearchManager;
import android.content.ComponentName;
import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatSpinner;
import androidx.appcompat.widget.SearchView;
import androidx.appcompat.widget.SeslSwitchBar;

import com.sec.sesl.tester.R;

import java.util.ArrayList;
import java.util.List;

import dev.oneuiproject.oneuiexample.ui.main.MainActivity;
import dev.oneuiproject.oneuiexample.ui.main.core.base.BaseFragment;

public class WidgetsFragment extends BaseFragment
        implements View.OnClickListener {

    SeslSwitchBar seslSwitchBar;

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        seslSwitchBar = ((MainActivity)requireActivity()).getDrawerLayout().getSwitchBar();

        int[] Ids = {R.id.fragment_btn_1,
                R.id.fragment_btn_2,
                R.id.fragment_btn_3,
                R.id.fragment_btn_4,
                R.id.fragment_btn_5};
        for (int id : Ids) view.findViewById(id).setOnClickListener(this);

        AppCompatSpinner spinner = view.findViewById(R.id.fragment_spinner);
        List<String> items = new ArrayList<>();
        for (int i = 1; i < 5; i++)
            items.add("Spinner Item " + i);
        ArrayAdapter<String> adapter = new ArrayAdapter<>(requireContext(),
                android.R.layout.simple_spinner_item, items);
        adapter.setDropDownViewResource(androidx.appcompat.R.layout.support_simple_spinner_dropdown_item);
        spinner.setAdapter(adapter);

        SearchView searchView = view.findViewById(R.id.fragment_searchview);
        SearchManager manager = (SearchManager) requireContext().getSystemService(Context.SEARCH_SERVICE);
        searchView.setSearchableInfo(manager.getSearchableInfo(
                new ComponentName(requireContext(), MainActivity.class)));
        searchView.seslSetUpButtonVisibility(VISIBLE);
        searchView.seslSetOnUpButtonClickListener(this);

    }

    private SeslSwitchBar.OnSwitchChangeListener listener = (v, isChecked) ->{
        seslSwitchBar.setProgressBarVisible(true);
        seslSwitchBar.postDelayed(() -> seslSwitchBar.setProgressBarVisible(false), 3_000);
    };

    @Override
    public void onHiddenChanged(boolean hidden) {
        try {
            if (!hidden) {
                seslSwitchBar.addOnSwitchChangeListener(listener);
            } else {
                seslSwitchBar.removeOnSwitchChangeListener(listener);
            }
        }catch (Exception ignore){}
    }

    @Override
    public int getLayoutResId() {
        return R.layout.sample3_fragment_widgets;
    }

    @Override
    public int getIconResId() {
        return dev.oneuiproject.oneui.R.drawable.ic_oui_game_launcher;
    }

    @Override
    public CharSequence getTitle() {
        return "Widgets";
    }

    @Override
    public boolean showSwitchBar() {
        return true;
    }

    @Override
    public void onClick(View v) {
        // no-op
    }

}

File: sample-app/src/main/java/dev/oneuiproject/oneuiexample/ui/main/fragments/widgets/WidgetsFragment.kt
package dev.oneuiproject.oneuiexample.ui.main.fragments.widgets

import android.os.Build
import android.os.Bundle
import android.view.View
import androidx.annotation.RequiresApi
import androidx.appcompat.widget.SeslSwitchBar
import androidx.appcompat.widget.SwitchCompat
import androidx.core.view.isVisible
import com.google.android.material.appbar.model.ButtonModel
import com.google.android.material.appbar.model.SuggestAppBarModel
import com.google.android.material.appbar.model.view.SuggestAppBarView
import com.sec.sesl.tester.R
import com.sec.sesl.tester.databinding.FragmentWidgetsBinding
import dev.oneuiproject.oneui.ktx.seslSetFillHorizontalPaddingEnabled
import dev.oneuiproject.oneui.ktx.setEntries
import dev.oneuiproject.oneui.ktx.setSearchableInfoFrom
import dev.oneuiproject.oneui.layout.DrawerLayout
import dev.oneuiproject.oneuiexample.ui.main.MainActivity
import dev.oneuiproject.oneuiexample.ui.main.core.base.AbsBaseFragment
import dev.oneuiproject.oneuiexample.ui.main.core.util.autoCleared
import dev.oneuiproject.oneuiexample.ui.main.core.util.semToast

class WidgetsFragment : AbsBaseFragment(R.layout.fragment_widgets) {

    private val binding by autoCleared { FragmentWidgetsBinding.bind(requireView()) }
    private var drawerLayout: DrawerLayout? = null

    private val listener by lazy {
        SeslSwitchBar.OnSwitchChangeListener { v: SwitchCompat?, isChecked: Boolean ->
            drawerLayout?.switchBar?.apply {
                setProgressBarVisible(true)
                postDelayed({ setProgressBarVisible(false) }, 3000)
            }
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        drawerLayout = (requireActivity() as MainActivity).drawerLayout

        binding.fragmentSpinner.apply {
            val entries = listOf("Item 1", "Item 2", "Item 3", "Item 4")
            setEntries(
                entries = entries,
                onSelect = { position, _ ->
                    if (this@WidgetsFragment.isResumed) {
                        position?.let { semToast("${entries[it]} clicked!") }
                    }
                }
            )
        }

        binding.searchview.apply {
            setSearchableInfoFrom(requireActivity())
            seslSetUpButtonVisibility(View.VISIBLE)
            seslSetOnUpButtonClickListener { semToast("SearchView Up button clicked!") }
        }

        binding.nsv.apply {
            seslSetFillHorizontalPaddingEnabled(true)
            seslSetGoToTopEnabled(true)
        }

        binding.bottomTip.setOnLinkClickListener {
            semToast("More details clicked!")
        }
    }

    override fun onStart() {
        super.onStart()
        drawerLayout?.apply {
            switchBar.apply { isVisible = true; addOnSwitchChangeListener(listener) }
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                setAppBarSuggestView(createSuggestAppBarModel())
            }
        }
    }

    override fun onStop() {
        super.onStop()
        drawerLayout?.apply {
            switchBar.apply { isVisible = false; removeOnSwitchChangeListener(listener) }
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                setAppBarSuggestView(null)
            }
        }
    }

    @RequiresApi(Build.VERSION_CODES.M)
    private fun createSuggestAppBarModel(): SuggestAppBarModel<SuggestAppBarView> =
        SuggestAppBarModel.Builder(requireContext()).apply {
            setTitle("This is an a suggestion view")
            setCloseClickListener {_, _ -> drawerLayout?.setAppBarSuggestView(null) }
            setButtons(
                arrayListOf(
                    ButtonModel(
                        text = "Action Button",
                        clickListener = { _, _ ->
                            semToast("Action button clicked!")
                        }
                    )
                ))
        }.build()

}